### DISCUSSION TITLE
Feature Request: Please add a CLI flag -config "key=value"

### DISCUSSION THREAD

**Discussion by marhar** [duckdb/duckdb#14214] [Ideas]
(note: updated to reflect comments by Tisj and carlopi.)

Please add a CLI flag `--config 'key=value'`.

- supports all values in `duckdb_settings()`.
- can be repeated; each --config supports setting one value.

## Context

In our CI/CD pipeline, we freshly install duckdb and several extensions into a TEMPDIR.
It would be good if we could set the extensions directory on the command line instead
of by `SET extension_directory = ...`.

Our current workaround is to run

```
HOME=$TEMPDIR duckdb
```

**Comment by Tishj** [comment]
`~/.duckdbrc` can be used to set settings like this on startup for every session

**Reply by jimdickson007** [reply]
@Tishj Are you able to point me at some documentation about ~/.duckdbrc  ?
Is this for Linux only ? Any equivalent in Windows land ?

**Reply by Tishj** [reply]
https://github.com/duckdb/duckdb/blob/1bd502e74908f1e6410b395c7bc9e768778f909b/tools/shell/shell.c#L19409

Should work on windows, have a look through this method to figure out where it's expecting the `.duckdbrc` file to live

**Comment by carlopi** [comment]
One though I had is whether this could be solved more generally by a `--config` parameter that takes a dictionary of values, I think that might have general sense (and resemble what other APIs do).

As @Tishj, you can get away with this via `~/.duckdbrc`

**Reply by Tishj** [reply]
I think that might make sense, I wonder how that would look in a CLI application though

`--config "key=value;key=value;key=value"` ?

there has to be some way to split the keyvalue pairs, or `--config` should be used to supply one keyvalue pair at a time
`--config "key=value" --config "key=value"`

**Reply by carlopi** [reply]
Second is probably easier & composable.

**Reply by marhar** [reply]
This is a great idea!  I will update the request to reflect this more general request.
